### PR TITLE
Remove superfluous leading backslash, closes 4520

### PR DESCRIPTION
**Pull request by ktomk**
In string context, there is no need to prefix the exception class-name
of the exception with any backslashes as these are always in the global
name-space.

Looks like the prefix is only useful for verbatim class names which is not
addressed in this issue (see #4068 for more details on verbatim class-
names and also a generic resolution suggestion for many fixers).

Additionally minor changes regarding standard (Phpstorm) inspections.

Closes: #4520
Related: #4068

### CODE REVIEW

**Comment by ktomk** [comment]
Next to the original request, I again need assistance w/ the fabbot. It paints the build red however, the code changes suggested are not suggested by project QA tools.

File: src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php [hunk 1]
```diff
@@ -183,6 +184,7 @@ private function fixPhpUnitClass(Tokens $tokens, $startIndex, $endIndex)
         $tokensAnalyzer = new TokensAnalyzer($tokens);
 
         for ($i = $endIndex - 1; $i > $startIndex; --$i) {
+            // @noinspection NotOptimalIfConditionsInspection
```
Comment (julienfalque) [inline src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php]: :-1: for adding tools instructions in the code.

File: src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php [hunk 2]
```diff
@@ -130,7 +131,7 @@ public function isRisky()
     /**
      * {@inheritdoc}
      */
-    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
```
Comment (julienfalque) [inline src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php]: Please revert unrelated changes. They should be applied globally in dedicated PRs.

File: src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php [hunk 3]
```diff
@@ -245,9 +247,10 @@ private function fixPhpUnitClass(Tokens $tokens, $startIndex, $endIndex)
             $tokens[$docBlockIndex] = new Token([T_DOC_COMMENT, $doc->getContent()]);
             $tokens->insertAt($braceIndex + 1, $newMethods);
 
-            $tokens[$braceIndex + $newMethods->getSize() + 1] = new Token([
+            $i1 = $braceIndex + $newMethods->getSize() + 1;
```
Comment (julienfalque) [inline src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php]: Please rename the variable (e.g. `$whitespaceIndex`), `$i1` does not carry any useful information about the data is holds.

File: src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php (hunk 1 above)
Comment (ktomk) [reply src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php]: Right, this does not look good for the code, removed.

File: src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php (hunk 3 above)
Comment (ktomk) [reply src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php]: Right, even used in a loop, all the other indexes are more verbosely named, it was only the $i below which might have irritated me a little.

Please consider your comment addressed and thanks for your thoughtful direction.

File: src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php (hunk 2 above)
Comment (ktomk) [reply src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php]: This must have been my muscle memory on the spot, you say it totally right this change has nothing at all to do with the issue at hand and has been removed from the change-set.

Please consider this comment addressed.

File: src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php [hunk 4]
```diff
@@ -275,12 +276,12 @@ private function extractContentFromAnnotation(Annotation $annotation)
...
         $params = [];
-        $exceptionClass = '\\'.ltrim($annotations['expectedException'], '\\');
+        $exceptionClass = ltrim($annotations['expectedException'], '\\');
 
         if ($this->configuration['use_class_const']) {
-            $params[] = $exceptionClass.'::class';
+            $params[] = '\\'.$exceptionClass.'::class';
         } else {
-            $params[] = "'{$exceptionClass}'";
+            $params[] = "'${exceptionClass}'";
```
Comment (julienfalque) [inline src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php]: This change should be reverted as well.

Comment (ktomk) [reply src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php]: @julienfalque This is actually what PHP-CS-Fixer applied after I have changed that line. I also was puzzled by this, could it be that the fixer suggests a different bracket format if there is no bracket format in use?

I removed it from the change-set in any case, so please consider your comment addressed.

File: src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php [hunk 5]
```diff
@@ -275,10 +276,10 @@ private function extractContentFromAnnotation(Annotation $annotation)
     private function annotationsToParamList(array $annotations)
     {
         $params = [];
-        $exceptionClass = '\\'.ltrim($annotations['expectedException'], '\\');
+        $exceptionClass = ltrim($annotations['expectedException'], '\\');
 
         if ($this->configuration['use_class_const']) {
-            $params[] = $exceptionClass.'::class';
+            $params[] = '\\'.$exceptionClass.'::class';
```
Comment (kubawerlos) [inline src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php]: Can we have `sprintf` here?

Comment (julienfalque) [reply src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php]: I would even write `"\\{$exceptionClass}::class"`.

Comment (ktomk) [reply src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php]: If you can find a resolution would be great. I can't write both in there at the same time.

IMHO neither of the three is of any difference. The projects PHP-CS-Fixer configuration suggests `"\\${exceptionClass}::class"`for string interpolation, but only unless the variant @julienfalque suggested is used.

So if you come to a conclusion, I'm happy to file it:

* A; `'a'.$b.'c'` style (as originally filed) 
* B: sprintf() call
* C: `"{$a}efgh"` style
* D: `"${a}efgh"` style

From the existing style in the file I prefer C, but please let me know which is the general preference in the project as I think this is worth to have at hand directly.

Comment (ktomk) [reply src/Fixer/PhpUnit/PhpUnitNoExpectationAnnotationFixer.php]: Myself concludes as even after a day, no result of the poll was given by others, that the resolution C is acceptable for this PR.

Thanks a lot for putting your efforts into the review, I really like to have multiple reviewers on a pull request and I know that this is nothing to expect.

**Comment by SpacePossum** [comment]
Thank you @ktomk.

**Comment by ktomk** [comment]
@SpacePossum: You're welcome. I'm glad I could be of help.
